compliance_checker/cf/cf_1_7.py

```diff
@@ -741,13 +741,11 @@ def check_grid_mapping(self, ds):
                 test_ctx.messages.append("The grid_mapping attribute {} "
                 "is not a string".format(item.grid_mapping))
                 test_ctx.out_of += 1
-                if ' ' in item.grid_mapping:  
-                    for x in item.grid_mapping.replace(":", '').split():      
-                        grid_mapping_variable_list.append(x)   
-                else:
-                    # [2/9] Note that in its simplest form the attribute comprises just 
-                    # a grid_mapping_name as a single word.
-                    grid_mapping_variable_list.append(item.grid_mapping)
+                
+                # [2/9] Note that in its simplest form the attribute comprises just 
+                # a grid_mapping_name as a single word.
+                grid_mapping_variable_list.append(regex.findall(r"(\w+)",item.grid_mapping))
```

Comment:
This still falls through due to `if` clause above.  If it's not a string, it will fail out of the function when attempting to match against a non-string object.  If it is a string, the `regex section will never be reached.  Perhaps put in an `else` clause or use a `continue` statement. If you're trying to *just* get the `grid_mapping` variable, you might try `regex.match(r"\w+")` since it implicitly anchors to start of string and won't fetch anything at the comma or beyond, so will skip coordinate variables in the latter form.

Reply:
added the else clause .... I missed it ! it was in the logic before I made the change